src/main/java/com/aws/greengrass/logmanager/util/FileHashGenerator.java

```diff
@@ -0,0 +1,63 @@
+package com.aws.greengrass.logmanager.util;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.logmanager.LogManagerService;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+import static com.aws.greengrass.util.Digest.calculate;
+
+public final class FileHashGenerator {
+    private static final Logger logger = LogManager.getLogger(LogManagerService.class);
+    // private constructor to make checkstyle happy
+    private FileHashGenerator() {
+        
+    }
+
+    /**
+     * calculate the hash for the log file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, the hash is null.
+     * @param logFile
+     * @return the nullable calculated hash for active file
+     */
+    public static Optional<String> calculateHashForLogFile(File logFile, int DEFAULT_LINES_FOR_DIGEST_NUM) {
+        Optional fileHash = Optional.ofNullable(null);
+        // empty file, then do not read
+        if (!logFile.isFile())
+            return fileHash;
+        try (BufferedReader r = Files.newBufferedReader(logFile.toPath(), StandardCharsets.UTF_8)) {
+            StringBuilder data = new StringBuilder();
+            int linesReadForDigest = 0;
+            String oneLine;
+            // read target number of lines; read one more line to prevent incomplete last line
+            // Therefore, there must be DEFAULT_LINES_FOR_DIGEST_NUM + 1 lines in each file.
+            while (linesReadForDigest <= DEFAULT_LINES_FOR_DIGEST_NUM) {
```

Comment:
This could be split into 2 methods:

1) Get lines from a file
2) Hash the content of the lines

Reply:
Also this could be improved a bit

This method would just read lines 
```java
List<String> linesNeeded = new ArrayList();

try (BufferedReader reader = Files.newBufferedReader(logFile.toPath(), StandardCharsets.UTF_8)) {
     String line = reader.readLine();

     if (line == null) {
        return linesNeeded;
     }

     linesNeeded.append(line);

      while (line != null && linesNeeded.size() < DEFAULT_LINES_FOR_DIGEST_NUM) {
          line = reader.readLine();
          if (line != null) {
              linesNeeded.append(line);
          }
     }
}

return linesNeeded;
```

Then `calculateHashForLogFile` calls it gets the lines and runs the logic to check if it can return a valid hash